platformics/graphql_api/files.py

```diff
@@ -325,15 +330,36 @@ def generate_multipart_upload_token(
 async def mark_upload_complete(
     file_id: strawberry.ID,
     principal: Principal = Depends(require_auth_principal),
-    cerbos_client: CerbosClient = Depends(get_cerbos_client),
+    authz_client: AuthzClient = Depends(get_authz_client),
     session: AsyncSession = Depends(get_db_session, use_cache=False),
     s3_client: S3Client = Depends(get_s3_client),
 ) -> db.File:
     """
     Once a file is uploaded, the front-end should make a markUploadComplete mutation
     to mark the file as ready for pipeline analysis.
     """
-    query = get_resource_query(principal, cerbos_client, CerbosAction.UPDATE, db.File)
+
+    # Get the type of entity that the file is related to
+    try:
+        file_row = (await session.execute(sa.select(db.File).where(db.File.id == file_id))).scalars().one()
+    except NoResultFound:
+        raise PlatformicsError("Unauthorized: cannot update file") from None
+
+    # Fetch the entity if have access to it
+    entity_class, entity = await get_entity_by_id(
```

Comment:
This call to get_entity_by_id checks to see whether the user has access to update the associated entity.

Reply:
If the get_resource_query() call below applies authz policies properly, we may not need this call!